kornia/feature/sold2/backbones.py

```diff
@@ -0,0 +1,436 @@
+"""Implements several backbone networks."""
+
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.functional import pixel_shuffle, softmax
+
+# [Hourglass backbone classes]
+
+
+class HourglassBackbone(nn.Module):
+    """Hourglass network, taken from https://github.com/zhou13/lcnn.
+
+    Args:
+        input_channel: number of input channels.
+        depth: number of residual blocks per hourglass module.
+        num_stacks: number of hourglass modules stacked together.
+        num_blocks: number of layers in each residual block.
+        num_classes: number of heads for the output of a hourglass module.
+    """
+
+    def __init__(self,
+                 input_channel: Optional[int] = 1,
```

Comment:
```suggestion
                 input_channel: int = 1,
```
`Optional` is only used with `None`